### DISCUSSION TITLE
Relation two to many schema

### DISCUSSION THREAD

**Discussion by dziugas-liaudinskas** [prisma/prisma#14892] [Q&A] [answered]
I know that two to many relation does not exist but I want to implement something like this and I'm stuck. 
I'm trying to create a prisma schema with this kind of relation - I have two models: Player and Game. Every game is "played" by two players and I'm trying to connect them with something like this:
```
model Player {
    id                  String @id @default(cuid())
    name                String
    games               Game[] 
}

model Game {
    id                  Int @id @default(autoincrement())
    player_1            Player
    player_2            Player
    winnerId            String
}
```
In this case I get an error which says `Ambiguous relation detected. The fields player_1 and player_2 in model Game both refer to Player. Please provide different relation names for them by adding @relation(<name>).`

And if I add relation with names which looks like this:
```
player_1            Player @relation(name: "GamePlayer1")
player_2            Player @relation(name: "GamePlayer2")
```
I get an error `The relation field player_1/player_2 on Model Game is missing an opposite relation field on the model Player.` I don't think that adding additional fields to Player model is necessary and also don't think that adding many to many relation which would look like this `players: Player[]` is smart.

What would be the right way to connect those two models based on my example?

**Comment by nurul3101** [comment, marked as answer]
Hi @dziugas-liaudinskas 👋 
You would need to model your schema file like this:
```prisma
model Player {
  id            String @id @default(cuid())
  name          String
  GameAsPlayer1 Game[] @relation(name: "player_1")
  GameAsPlayer2 Game[] @relation(name: "player_2")
}

model Game {
  id        Int    @id @default(autoincrement())
  player_1  Player @relation(name: "player_1", fields: [player1Id], references: [id])
  player_2  Player @relation(name: "player_2", fields: [player2Id], references: [id])
  winnerId  String
  player1Id String
  player2Id String
}
```
Also, please note that the relation fields `GameAsPlayer1` and `GameAsPlayer2` wouldn't exist on the database

**Reply by dziugas-liaudinskas** [reply]
Thanks!

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
